docs/development/designs/automationGo.md

```diff
@@ -0,0 +1,49 @@
+# Problem
+
+Maintaining unsupported versions of Kubernetes and by extention Golang/Go requires large amounts of monitoring, evaluation, occassional deep dives into the functionality and differences in versions, and large amounts of repetitive searchable documentation for the Open-Source Community and transparency.
+
+Specifically for patching golang, we are required to monitor their announcements for vulnerabilities, open our own CVE issues for backports that are consistent to allow for easily searchable auditing and transparency for the open-source community, attempt backporting the fixes, and when the backport doesn't apply cleanly, discover the differences in versions to make the backport apply, if applicable.
+
+While the last task will require effort by the team, the rest of the task can be automated to provide clear and consistent [unknown phrasing/word here]. An automated solution would build a consistant audit trail for which CVEs have been patched or why they weren't, any public discussion on the decision process, and many other data points that could be of interest.
+
+# Solution Overview
+
+I propose creating a go application for patching. Starting with Golang and potentially extending the functionality to Kubernetes as the workflow should be similar.
+
+## Golang Patching Runbook
+
+1. Golang release CVE announcement
+2. Open cve backport issues for EKS supported versions
+3. Attempt cherry-pick
+   3a. Find upstream commit fix
+   3b. git cherry-pick {commit hash}
+   3c. fix any merge issues
+4. Create .patch file with EKS patch header
+5. Create pr with .patch file fix and updated .spec file
+   (optional) upload additional source files required for patch
+6. Presubmit: pass/fail
+   6a (pass). Merge PR.
+   6b (fail). Repeat steps 3c - 6 with attempted fixes for build failures
+
+## Expected Kubernetes Patching Runbook
+
+Similar to Golang
+
+## Deep Dive
+
+Use a prow job to a call the CVE search Golang repo for CVE [backport] issue. Once CVE backport issue is created, trigger a job to open AWS/EKS-Distro-Build-Tooling issues for eks supported versions of go. Once these issues are created, watch the upstream issue for closed by [commit] comment. Use this commit to cherry-pick, format-patch, and submit initial PR is no merge conflicts. If merge conflict, create comment in the issue that a pr couldn't be autogenerated.
+
+## Tools
+
+Cobra cli to make easy calls and pass [options]/flags.
+Add to the existing scaffolding under eks-distro-build-tooling/tools/cmd/eksGoTool.
```

Comment:
so just to clarify this, and I think we're on the same page already, we should add the invocation and go-specific stuff to `eksGoTool`, and all the generic functionality to `eks-distro-build-tooling/tools/pkg`

Reply:
Correct. This would allow for future projects to be able to have their own tool without needing to duplicate the generic functionality.